### ISSUE TITLE
Better way to close the log file?

### ISSUE THREAD

**Issue by tylermontneyacc** [PowershellFrameworkCollective/psframework#562]
```powershell
# Open log file    
$paramSetPSFLoggingProvider = @{
        Name         = 'logfile'
        InstanceName = $TaskName
        FilePath     = "$LogPath$TaskName.log"
        FileType     = 'CMTrace'
        Enabled      = $true
    }
    Set-PSFLoggingProvider @paramSetPSFLoggingProvider -ErrorAction Stop

# Do work

# Close log file
$paramSetPSFLoggingProvider = @{
    Name         = 'logfile'
    InstanceName = $TaskName
    FilePath     = "$LogPath$TaskName.log"
    FileType     = 'CMTrace'
    Enabled      = $false
}
Set-PSFLoggingProvider @paramSetPSFLoggingProvider
```

If I don't call `Set-PSFLoggingProvider` again, as shown, the log files are locked. Is there a proper way to "dispose" once finished, or do i even need to worry?

**Comment by FriedrichWeinmann** [comment]
The key command to closing out is `Wait-PSFMessage` which ensures all logs are flushed (Theoretically it should auto-flush, but thanks to the way PowerShell handles the process termination events it won't always do so).
You only need to worry about disabling it - after flushing - if your process is going to keep running - e.g. when running a script from your console, but keeping the console open afterwards.

To disable it, you do not include the rest of the configuration:

```powershell
$paramSetPSFLoggingProvider = @{
    Name         = 'logfile'
    InstanceName = $TaskName
    Enabled      = $false
}
Set-PSFLoggingProvider @paramSetPSFLoggingProvider
```

That said, there is probably some value in providing a `Disable-PSFLoggingProvider` command that combines disabling and waiting (and have the waiting be specific to the logging provider, not a global wait).

**Comment by tylermontneyacc** [comment]
- So, then re-running `Set-PSFLoggingProvider` would be more reliable for the time being?
- I agree, providing `Disable-PSFLoggingProvider` would be good. It makes command discovery a lot easier when there's a corresponding verb.

**Comment by FriedrichWeinmann** [comment]
`Disable-PSFLoggingProvider` has been released with v1.7.270
That should cover proper disabling in the most convenient manner
